### PR TITLE
Fix for rc_env_allow not allowing * which is the intended behaviour

### PR DESCRIPTION
**Pull request by neeravkumar**

### CODE REVIEW

**Comment by neeravkumar** [comment]
Could anybody comment if there are any issues with this pull request due to which it hasn't been merged for so long?

File: src/rc/rc-misc.c [hunk 1]
```diff
@@ -91,22 +89,26 @@ env_filter(void)
...
-		if (env_whitelist[i])
-			continue;
-
-		/* Check our user defined list */
-		if (rc_stringlist_find(env_allow, env->value))
-			continue;
-
-		/* OK, not allowed! */
-		unsetenv(env->value);
+	if (rc_conf_value("rc_env_allow") != "*") {
```
Comment (dwfreed) [inline src/rc/rc-misc.c]: You must use strcmp here

File: src/rc/rc-misc.c [hunk 2]
```diff
@@ -91,22 +89,26 @@ env_filter(void)
...
+			if (env_whitelist[i])
+			  continue;
+
+		  /* Check our user defined list */
+		  if (rc_stringlist_find(env_allow, env->value))
+			  continue;
+
+		  /* OK, not allowed! */
+		  unsetenv(env->value);
+		  }
```
Comment (dwfreed) [inline src/rc/rc-misc.c]: Please fix the indentation to be proper, consistent, and using tabs

**Comment by williamh** [comment]
There is a much simpler fix for this since we have rc_stringlist_find.
I have tested and will merge my fix shortly.
you will see it on this pull request.

**Pull request closed without merging**